realm/realm-library/src/objectServer/java/io/realm/internal/objectstore/OsMongoIterable.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.objectstore;
+
+import org.bson.codecs.configuration.CodecRegistry;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.realm.ObjectServerError;
+import io.realm.internal.jni.JniBsonProtocol;
+import io.realm.internal.jni.OsJNIResultCallback;
+import io.realm.internal.network.ResultHandler;
+
+public abstract class OsMongoIterable<ResultT> {
```

Comment:
Seems to remember some comment about only using `Os`-prefix if it actually wrapped an underlying OS object.

Reply:
I talked about it with @cmelchior and it's OK to use it for helper classes which don't instantiate objects - unless I misunderstood it @cmelchior?

Reply:
I think there was a bit of a misunderstanding then. Normally we only use `Os<X>` for classes wrapping some ObjectStore class. I believe we chatted about finalizers? 🤔 

So, in this case, it should probably be merged with the public MongoIterable

Reply:
Sure, I'll get it done.